src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -949,16 +963,18 @@ private unsafe bool TryGetOrCreateObjectForComInstanceInternal(
                 innerMaybe,
                 flags,
                 out IntPtr identity,
-                out IntPtr inner);
+                out IntPtr inner,
+                out IntPtr referenceTrackerMaybe);
 
             using ComHolder releaseIdentity = new ComHolder(identity);
+            using NullableComHolder referenceTracker = new NullableComHolder(referenceTrackerMaybe);
 
             // If the user has requested a unique instance,
             // we will immediately create the object, register it,
             // and return.
             if (flags.HasFlag(CreateObjectFlags.UniqueInstance))
             {
-                retValue = CreateAndRegisterObjectForComInstance(identity, inner, flags);
+                retValue = CreateAndRegisterObjectForComInstance(identity, inner, flags, ref *&referenceTracker);
```

Comment:
What is the `*&` construct trying to achieve?

Reply:
It's used to pass a `ref` to a variable in a using statement. C# doesn't let you do it, because `using` variables are "controlled mutability" (which would be sufficient, but there's no syntax to actually get a controlled mutability managed pointer). So this allows to just strip that and get a normal `ref` to it.

Reply:
Would it better to just use IntPtr and old-fashioned finally? No unsafe code needed.

Reply:
Updated!